From d07d50c6f8a5793badc50fb2c964b87dec87a3a1 Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/source-build/issues/1868)
From: Antoine Martin <dev@ayakael.net>
Date: Thu, 15 Jul 2021 15:37:33 +0000
Subject: [PATCH 1/1] Fixed darc init for Alpine

Darc has a segmentation fault on Alpine due to not chosing the correct
binary architecture. This patch deletes all the wrong ones so that it
is forced to chose the correct one. This issue only occurs when using 
dotnet-stage0-runtime, thus libunwind is suspected to be the issue as
stage0 wasn't built with system libunwind.
---
 eng/common/darc-init.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/eng/common/darc-init.sh b/eng/common/darc-init.sh
index 82b2b57..5227a37 100755
--- a/eng/common/darc-init.sh
+++ b/eng/common/darc-init.sh
@@ -77,6 +77,9 @@ function InstallDarcCli {
   else
     echo $($dotnet_root/dotnet tool install $darc_cli_package_name --version $darcVersion --add-source "$arcadeServicesSource" -v $verbosity --tool-path "$toolpath")
   fi
+  for i in  win* ubuntu.1* rhel* osx fedora* debian*; do
+    rm -R $toolpath/.store/microsoft.dotnet.darc/*/microsoft.dotnet.darc/*/tools/netcoreapp3.0/any/runtimes/${i}
+  done
 }
 
 InstallDarcCli
-- 
2.30.2

